Add descriptive commentary

KaiKerlaff opened the pull request:
Closes #69

KaiKerlaff committed:
Add descriptive commentary

olexandr-konovalov commented:
@KaiKerlaff many thanks - I had no chance to read all in all the details yet, and hope to do that after the next editing iteration, but overall it looks like a very good attempt of reviewing the code. Just some elaborations of what could be a good practice. In the current state, it's hard to see diffs, because large chunks of code are changed. It is hard to see whether the code is changed, or just the comment is changed, because of the use of the end-of-line comments. Also, it is hard to read comments since they are very long (not everyone uses automated wrapping up of long lines) and in the future it will be harder to see further diffs in long lines of comments. 

Therefore, I suggest to minimise the use of end-of-line comments, only keeping short ones, where they were before. Longer ones should be placed as separate blocks of comments, before the code they are referring to. The line width should be limited to 70-80 characters - use hard line breaks, if you have longer text.

I had a brief look at MATLAB Style Guidelines 2.0, available from https://uk.mathworks.com/matlabcentral/fileexchange/46056-matlab-style-guidelines-2-0 (you need a free MathWorks account to download the PDF). It may be useful to look at it or the cheatsheet at https://uk.mathworks.com/matlabcentral/fileexchange/45047-matlab-style-guidelines-cheat-sheet. In particular, it says:

> **Minimize use of end of line comments**.
The descriptiveness of end of line comments is constrained by the typical 80 column line length. In general they should only be used as an adjunct to variable declaration.

olexandr-konovalov commented:
@KaiKerlaff I have added a link to the description - first, it's useful to link issues and PRs with cross-references. Second, there are some keywords which will automatically close the corresponding issue when the PR is merged.

olexandr-konovalov commented:
@KaiKerlaff there are now conflicts after recent merges. This is one of the reasons why end-of-line comments are not so good - having them on separate lines would reduce the number on conflicts, but now you edited warp level in the code in one PR, and a comment in another...

olexandr-konovalov commented:
P.S. @KaiKerlaff of course, you don't have to use web editor as GitHub tells you, since you are now empowered by the knowledge of Git command line tricks :)

KaiKerlaff committed:
Update comments to reflect new implementation of warpFactorLevel

KaiKerlaff commented:
@olexandr-konovalov I have come up with a second draft and added it as a separate commit to this PR, as I think the old comments (in a9fa3b2) may be relevant for future reference. Please let me know if it is better practise to squash these commits, however.

While resolving the conflicts, I have also changed the comments to reflect the content of the new code. In particular, **I have not changed any code from 86d678b, only comments**. Thanks for your advice on end-of-line comments - I have limited the comment line length to 80 characters here, and comments are generally directly above the line they refer to :)

olexandr-konovalov commented:
@KaiKerlaff fine to keep different commits. We will have to merge it without rebasing, and as far as I can see, standard merge would have no conflicts, since your 2nd commit fixes them all.

olexandr-konovalov commented:
@KaiKerlaff could you resolve conflicts and then let's merge this before code diverges further.

KaiKerlaff committed:
Resolve conflicts

KaiKerlaff commented:
@olexandr-konovalov just resolved the conflicts now.

KaiKerlaff committed:
Merge branch 'dolphin-acoustics-vip:stable-1' into warp-comment

olexandr-konovalov commented:
@KaiKerlaff let's try to merge this - have you managed to resolve the conflicts?

KaiKerlaff commented:
@olexandr-konovalov yes, all conflicts are resolved now!

olexandr-konovalov commented:
@KaiKerlaff thanks - it says "This branch cannot be rebased due to conflicts" and it's actually what it is - it can not be rebased but it can be merged. I did it locally in the command line. Before I will push to this repository, could you please check that it all looks good in my fork:
- https://github.com/olexandr-konovalov/artwarp/blob/stable-1/warp.m
- https://github.com/olexandr-konovalov/artwarp/commit/eac4d24841ad0ddb33405881ee651695143e0a83

KaiKerlaff commented:
@olexandr-konovalov ah I see, thank you! And yes your fork looks correct, all code (excluding comments and whitespace) is identical to the stable-1 branch at https://github.com/dolphin-acoustics-vip/artwarp/blob/stable-1/warp.m.

olexandr-konovalov commented:
Thanks for checking @KaiKerlaff - I've now merged it locally, and the PR got automatically closed after the push.
